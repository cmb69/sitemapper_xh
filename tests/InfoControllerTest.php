<?php

namespace Sitemapper;

use ApprovalTests\Approvals;
use PHPUnit\Framework\TestCase;
use Plib\FakeRequest;
use Plib\FakeSystemChecker;
use Plib\View;

class InfoControllerTest extends TestCase
{
    public function testExecute(): void
    {
        $model = $this->createStub(Model::class);
        $model->method("installedLanguages")->willReturn(["en", "de"]);
        $view = new View("./views/", XH_includeVar("./languages/en.php", "plugin_tx")["sitemapper"]);
        $sut = new InfoController(
            "./",
            "en",
            "./plugins/sitemapper",
            $model,
            new FakeSystemChecker(),
            $view
        );
        Approvals::verifyHtml($sut->execute(new FakeRequest()));
    }
}
