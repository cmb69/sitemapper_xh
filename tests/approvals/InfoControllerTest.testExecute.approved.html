<div>
  <h1>Sitemapper 3.3-dev</h1>
  <h2>Available Sitemaps</h2>
  <ul>
    <li>
      <a href="/?sitemapper_index" target="_blank">index</a>
    </li>
    <li>
      <a href="/?sitemapper_sitemap" target="_blank">en</a>
    </li>
    <li>
      <a href="/de/?sitemapper_sitemap" target="_blank">de</a>
    </li>
  </ul>
  <h2>System check</h2>
  <p class="xh_fail">PHP version ≥ 7.1.0</p>
  <p class="xh_fail">CMSimple_XH version ≥ 1.7.0</p>
  <p class="xh_fail">Plib_XH version ≥ 1.4</p>
  <p class="xh_warn">'./plugins/sitemapper/config/' writable</p>
  <p class="xh_warn">'./plugins/sitemapper/languages/' writable</p>
</div>
